cmd/s3-peer-client.go

```diff
@@ -187,9 +142,7 @@ func (s3p *s3Peers) SendRPC(peers []string, method string, args interface {
 // request to all peers. Currently we log an error and continue.
 func S3PeersUpdateBucketNotification(bucket string, ncfg *notificationConfig) {
 	setBNPArgs := &SetBNPArgs{Bucket: bucket, NCfg: ncfg}
-	peers := globalS3Peers.GetPeers()
-	errsMap := globalS3Peers.SendRPC(peers, "S3.SetBucketNotificationPeer",
-		setBNPArgs)
+	errsMap := globalS3Peers.SendUpdate(globalS3Peers.Peers, setBNPArgs)
```

Comment:
> In the case where some peers return errors, we could have a policy of doing a controlled number of retries (only on the failing peers). SendUpdate has this function type to allow for this type of retry - we need to only pass the peers that had failures. This is also the reason it returns an error map.

SendUpdate can return a slice of errors and still achieve what map can achieve and SendUpdate() function becomes simpler. a `nil` error in the slice indicates that the corresponding peer succeeded, `non-nil` indicates error.